test_integration/test_ee_short_region_slop.py

```diff
@@ -325,11 +364,135 @@ def extract_data_region(report, cols, key):
+        plt.errorbar(xdata, ydata, ydata_std, fmt='.')
+        if ylim is not None:
+            plt.ylim(ylim)
+    plt.title('{} region {}'.format(region_type, yaxis))
+    plt.xlabel(xaxis)
+    plt.ylabel(yaxis)
+    plt.legend(('fixed', 'dynamic'))
+
+
+def read_trace(path):
+    converters={'REGION_HASH': lambda xx: int(xx, 16)}
+    return pandas.read_csv(path, sep='|', comment='#', converters=converters)
+
+
+def get_ipc(trace):
+    di = trace['INSTRUCTIONS_RETIRED'].diff()
+    dc = trace['CYCLES_THREAD'].diff()
+    return di / dc
+
+def get_freq(trace):
+    return trace['FREQUENCY']
+
+def get_relative_time(trace):
+    return trace['TIME'] - trace['TIME'].iloc[0]
+
+
+def get_uncore_freq(trace):
+    return trace['MSR::UNCORE_PERF_STATUS:FREQ']
+
+
+def select_region_count(trace, region_count):
+    return trace.loc[trace['REGION_COUNT'] == region_count]
+
+
+def select_region(trace, region_prefix, extension=''):
+    """e.g.:
+    >>> select_region(trace, 'scaling_2')
+    >>> select_region(trace, 'scaling_', range(12))
+    >>> select_region(trace, '', ['scaling_3', 'timed_3'])
+
+    """
+    if type(extension) is str:
+        extension = [extension]
+    region_hash = []
+    for ext in extension:
+        region_name = '{}{}'.format(region_prefix, ext)
+        region_hash.append(geopmpy.hash.crc32_str(region_name))
+    return trace.loc[trace['REGION_HASH'].isin(region_hash)]
+
+
+def select_time_range(trace, duration, begin_time=None):
+    if begin_time is None:
+        begin_time = trace['TIME'].iloc[0]
+    end_time = begin_time + duration
+    return trace.loc[(trace['TIME'] >= begin_time) &
+                     (trace['TIME'] < end_time)]
+
+
+def generate_trace_region_plot(trace, out_path, region_name, region_count):
+    if type(region_count) is int:
+        region_count = [region_count]
+    ipc_legend = []
+    freq_legend = []
+    plt.figure(figsize=(11,11))
+    plt.title(region_name)
+    for rc in region_count:
+        selected_trace = select_region(trace, region_name)
+        selected_trace = select_region_count(selected_trace, rc)
+        ipc = get_ipc(selected_trace)
+        freq = get_freq(selected_trace)
+        time = get_relative_time(selected_trace)
+        if time.iloc[0] != 0.0:
+            sys.stdout.write('time[0] = {}'.format(time.iloc[0]))
+        plt.subplot(2, 1, 1)
+        plt.plot(time.iloc[:-1], ipc.iloc[1:], '.-')
+        plt.ylim(g_plot_ipc_lim)
+        plt.xlabel('time since region start (sec)')
+        plt.ylabel('IPC')
+        ipc_legend.append('count {}'.format(rc))
+        plt.subplot(2, 1, 2)
+        plt.plot(time.iloc[:-1], freq.iloc[1:], '.-')
+        plt.xlabel('time since region start (sec)')
+        plt.ylabel('CPU freq (Hz)')
+        plt.ylim(g_plot_freq_lim)
+        freq_legend.append('count {}'.format(rc))
+    plt.subplot(2,1,1)
+    plt.legend(ipc_legend)
+    plt.subplot(2,1,2)
+    plt.legend(freq_legend)
+    plt.savefig(out_path)
+    plt.close()
```

Comment:
What do you have against \n's ? 😆

Reply:
I have removed the patch that has the memory alignment change.  I tried to refactor the blob into a loop, but it wasn't worth the hassle.